Sources/CustomDump/Dump.swift

```diff
@@ -122,123 +144,132 @@ public func customDump<T, TargetStream>(
         dumpChildren(
-          of: associatedValuesMirror,
-          prefix: "\(child.label ?? "@unknown")(",
+          of: mirror,
+          prefix: "Set([", suffix: "])",
+          by: {
+            _customDump($0.value, name: nil, indent: 0, maxDepth: 1)
+            < _customDump($1.value, name: nil, indent: 0, maxDepth: 1)
+          })
+
+      case (_, .struct?):
+        dumpChildren(of: mirror, prefix: "\(typeName(mirror.subjectType))(", suffix: ")")
+
+      case (_, .tuple?):
+        dumpChildren(
+          of: mirror,
+          prefix: "(",
           suffix: ")",
           { child, _ in
             if child.label?.first == "." {
               child.label = nil
             }
-          }
-        )
-      } else {
-        out.write("\(value)")
-      }
-
-    case (_, .optional?):
-      if let value = mirror.children.first?.value {
-        customDumpHelp(value, to: &out, name: nil, indent: 0, maxDepth: maxDepth)
-      } else {
-        out.write("nil")
-      }
-
-    case (_, .set?):
-      dumpChildren(
-        of: mirror,
-        prefix: "Set([", suffix: "])",
-        by: {
-          _customDump($0.value, name: nil, indent: 0, maxDepth: 1)
-            < _customDump($1.value, name: nil, indent: 0, maxDepth: 1)
-        })
-
-    case (_, .struct?):
-      dumpChildren(of: mirror, prefix: "\(typeName(mirror.subjectType))(", suffix: ")")
-
-    case (_, .tuple?):
-      dumpChildren(
-        of: mirror,
-        prefix: "(",
-        suffix: ")",
-        { child, _ in
-          if child.label?.first == "." {
-            child.label = nil
-          }
-        })
+          })
 
-    default:
-      if let value = stringFromStringProtocol(value) {
-        if value.contains(where: \.isNewline) {
-          if maxDepth == 0 {
-            out.write("\"…\"")
+      default:
+        if let value = stringFromStringProtocol(value) {
+          if value.contains(where: \.isNewline) {
+            if maxDepth == 0 {
+              out.write("\"…\"")
+            } else {
+              let hashes = String(repeating: "#", count: value.hashCount)
+              out.write("\(hashes)\"\"\"")
+              out.write("\n")
+              print(value.indenting(by: name != nil ? 2 : 0), to: &out)
+              out.write(name != nil ? "  \"\"\"\(hashes)" : "\"\"\"\(hashes)")
+            }
           } else {
-            let hashes = String(repeating: "#", count: value.hashCount)
-            out.write("\(hashes)\"\"\"")
-            out.write("\n")
-            print(value.indenting(by: name != nil ? 2 : 0), to: &out)
-            out.write(name != nil ? "  \"\"\"\(hashes)" : "\"\"\"\(hashes)")
+            out.write(value.debugDescription)
           }
         } else {
-          out.write(value.debugDescription)
+          out.write("\(value)")
         }
-      } else {
-        out.write("\(value)")
+      }
+
+      if let superclassMirror = mirror.superclassMirror, value is CustomDumpIncludeSuperclass {
```

Comment:
If you remove this `value is CustomDumpIncludeSuperclass` check the failures will happen

Reply:
I pushed it here https://github.com/tahirmt/swift-custom-dump/tree/dump-improvements-failure